pkg/aws/services/tagging.go

```diff
@@ -0,0 +1,97 @@
+package services
+
+import (
+	"context"
+	"github.com/aws/aws-application-networking-k8s/pkg/utils"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	taggingapi "github.com/aws/aws-sdk-go/service/resourcegroupstaggingapi"
+	taggingapiiface "github.com/aws/aws-sdk-go/service/resourcegroupstaggingapi/resourcegroupstaggingapiiface"
+)
+
+//go:generate mockgen -destination tagging_mocks.go -package services github.com/aws/aws-application-networking-k8s/pkg/aws/services Tagging
+
+const (
+	resourceTypePrefix = "vpc-lattice:"
+
+	ResourceTypeTargetGroup = resourceTypePrefix + "targetgroup"
+
+	maxArnsPerGetResourcesApi = 100
+)
+
+type Tags = map[string]*string
+
+type Tagging interface {
+	taggingapiiface.ResourceGroupsTaggingAPIAPI
+
+	// Receives a list of arns and returns arn-to-tags map.
+	GetTagsFromArns(ctx context.Context, arns []*string) (map[string]Tags, error)
+
+	// Finds one resource that matches the given set of tags.
+	FindResourceWithTags(ctx context.Context, resourceType string, tags Tags) (*string, error)
```

Comment:
With -> By, also should return arn.ARN, not clear what *string means here

Reply:
resourceType string, can be strong typed, we have limited options better to keep it enum-like

Reply:
makes sense